### DISCUSSION TITLE
How to write json data into const char *

### DISCUSSION THREAD

**Discussion by adpathak** [nlohmann/json#3777] [Q&A] [answered]
Const char *xyz:
xyz=jsonObj["pcf"]["freediameter"].at["identiti"];
The above is not working.
What is the correct way to write in xyz?

**Comment by falbrechtskirchinger** [comment, marked as answer]
Assuming the JSON value is a string, try this:
```cpp
auto &&str = jsonObj["pcf"]["freediameter"].at["identiti"].get_ref<const std::string &>();
const char *xyz = str.c_str();
```

The lifetime of `xyz` is bound to the JSON value.

Alternatively:
```cpp
auto str = jsonObj["pcf"]["freediameter"].at["identiti"].get<std::string>();
const char *xyz = str.c_str();
```

The lifetime of `xyz` is bound to `str`.

**Reply by adpathak** [reply]
What is method for storing into arrary of struct?

**Reply by adpathak** [reply]
@falbrechtskirchinger ->i am getting this error, before that on same machine i am using #include "json.hpp" using json = nlohmann::json;(getting no error) but in different module i am getting
i am also including these -ljson-c -ljsoncpp but still getting the below error
fatal error: json.hpp: No such file or directory

**Reply by adpathak** [reply]
can u help me please?

**Reply by falbrechtskirchinger** [reply]
If you still need help with this issue, please open a new discussion with more information. I'm not sure what `json-c` and `jsoncpp` have to do with this.

**Comment by gregmarr** [comment]
Why do you want a `const char *` variable?  There are safer alternatives.  Are you passing this to a function that takes `const char *`?

**Reply by adpathak** [reply]
It is a struct member and I can't change that.

**Reply by gregmarr** [reply]
Oh, so really dangerous.  Does the struct take ownership of the pointer, and delete/free it when the struct is destroyed?  That makes things much more complicated.
